filters/auth/authclient.go

```diff
@@ -92,9 +94,39 @@ func (ac *authClient) doClonedGet(ctx filters.FilterContext) (int, error) {
 	return rsp.StatusCode, nil
 }
 
-func createHTTPClient(timeout time.Duration, quit chan struct{}, maxIdleConns int) (*http.Client, error) {
+type authClientDialer struct {
```

Comment:
@szuecs after on our offline discussion, I remembered that there is a way to make this more granular to see how long DNS resolution takes. What do you think about using https://golang.org/pkg/net/http/httptrace?

I think this would be valuable for the `proxy` operation as well.

You can find an example [here](https://github.com/opentracing-contrib/go-stdlib/blob/master/nethttp/client.go#L239).

Reply:
I will measure the impact, but sounds good.